src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -423,32 +423,61 @@ namespace AppInstaller::Repository::Microsoft::Schema::V1_4
         return result;
     }
 
-    std::optional<SQLite::rowid_t> DependenciesTable::IsValueReferenced(const SQLite::Connection& connection, std::string_view valueName, SQLite::rowid_t valueRowId)
+    bool DependenciesTable::IsValueReferenced(const SQLite::Connection& connection, std::string_view tableName, SQLite::rowid_t valueRowId)
     {
         StatementBuilder builder;
 
-        std::array<std::string_view, 3> dependenciesColumns =
-        { s_DependenciesTable_MinVersion_Column_Name,
-            s_DependenciesTable_Manifest_Column_Name,
-            s_DependenciesTable_PackageId_Column_Name };
+        if (tableName != V1_0::VersionTable::TableName())
+        {
+            return false;
+        }
+
+        std::array<std::string_view, 1> columns = { s_DependenciesTable_MinVersion_Column_Name };
+        bool referenced = false;
 
-        auto result = std::find(dependenciesColumns.begin(), dependenciesColumns.end(), valueName);
-        if (result == std::end(dependenciesColumns))
+        for(auto column: columns)
         {
-            THROW_HR(APPINSTALLER_CLI_ERROR_INVALID_TABLE_COLUMN);
+            builder.Select(SQLite::RowIDName).From(s_DependenciesTable_Table_Name).Where(column).Equals(Unbound).Limit(1);
```

Comment:
nit: You can pass `valueRowId` instead of `Unbound` to achieve the same thing here.  Not need to change it, just a note in case you weren't aware.